diff --git a/js/private/js_binary.sh.tpl b/js/private/js_binary.sh.tpl
index f0e87a6f..cfdb4a51 100755
--- a/js/private/js_binary.sh.tpl
+++ b/js/private/js_binary.sh.tpl
@@ -99,9 +99,13 @@ function logf_debug {
 }
 
 function resolve_execroot_bin_path {
     local short_path="$1"
+    local bindir="${BAZEL_BINDIR:-$JS_BINARY__BINDIR}"
+    if [ "${JS_BINARY__USE_EXECROOT_ENTRY_POINT:-}" ]; then
+        bindir="$JS_BINARY__BINDIR"
+    fi
     if [[ "$short_path" == ../* ]]; then
-        echo "$JS_BINARY__EXECROOT/${BAZEL_BINDIR:-$JS_BINARY__BINDIR}/external/${short_path:3}"
+        echo "$JS_BINARY__EXECROOT/$bindir/external/${short_path:3}"
     else
-        echo "$JS_BINARY__EXECROOT/${BAZEL_BINDIR:-$JS_BINARY__BINDIR}/$short_path"
+        echo "$JS_BINARY__EXECROOT/$bindir/$short_path"
     fi
 }
